spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/MqttPahoMessageHandler.java

```diff
@@ -208,7 +204,7 @@ private void sendDeliveryComplete(IMqttDeliveryToken token) {
 	}
 
 	@Override
-	public void connectionLost(Throwable cause) {
+	public synchronized void connectionLost(Throwable cause) {
```

Comment:
Would you mind to explain a bit deeper how this `synchronized` fix helps ?
Thanks in advance